server/datastore/mysql/software_installers.go

```diff
@@ -10,6 +10,56 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
+func (ds *Datastore) ListPendingSoftwareInstallDetails(ctx context.Context, hostID uint) ([]string, error) {
+	const stmt = `
+  SELECT
+    hsi.execution_id
+  FROM
+    host_software_installs hsi
+  WHERE
+    hsi.host_id = ?
+  AND
+    hsi.install_script_exit_code IS NULL
+  AND
+    hsi.pre_install_query_output IS NULL
+`
+	var results []string
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &results, stmt, hostID); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "list pending software installs")
+	}
+	return results, nil
+}
+
+func (ds *Datastore) GetSoftwareInstallDetails(ctx context.Context, executionId string) (*fleet.SoftwareInstallDetails, error) {
+	const stmt = `
+  SELECT
+    hsi.host_id AS host_id,
+    hsi.execution_id AS execution_id,
+    hsi.software_installer_id AS installer_id,
+    si.pre_install_query AS pre_install_condition,
+    inst.contents AS install_script,
+    pisnt.contents AS post_install_script
+  FROM
+    host_software_installs hsi
+  INNER JOIN
+    software_installers si
+    ON hsi.software_installer_id = si.id
+  LEFT OUTER JOIN
+    script_contents inst
+    ON inst.id = si.install_script_content_id
+  LEFT OUTER JOIN
+    script_contents pisnt
+    ON pisnt.id = si.post_install_script_content_id
+  WHERE
+    hsi.execution_id = ?`
+
+	result := &fleet.SoftwareInstallDetails{}
+	if err := sqlx.GetContext(ctx, ds.reader(ctx), result, stmt, executionId); err != nil {
```

Comment:
When using `GetContext`, it will fail if there is no row found with `sql.ErrNoRows`. Typically we catch that error and convert it to a `notFound` error that results in a 404 at the API level (e.g. https://github.com/fleetdm/fleet/blob/0c0b981d9f1e59360cafcabb30de0988df48cd5b/server/datastore/mysql/software_installers.go#L117).